tests/GridFS/BucketFunctionalTest.php

```diff
@@ -814,6 +814,10 @@ public function testUploadingFirstFileCreatesIndexes(): void
 
     public function testExistingIndexIsReused(): void
     {
+        // The collections may exist from other tests, ensure they are removed before and after the test
+        $this->dropCollection($this->getDatabaseName(), 'fs.chunks');
+        $this->dropCollection($this->getDatabaseName(), 'fs.files');
+
```

Comment:
I added a comment and assertions on index existence.

Reply:
Oof, trying to think back to it. I believe the issue was that the shell used slightly different index settings, resulting in valid indexes that weren't recognised. It may have been an issue with using a double for the index direction instead of an int. I don't remember precisely though, indicating that I should've left a comment at the time.